sot/opcode_translator/instruction_utils/instruction_utils.py

```diff
@@ -325,3 +325,24 @@ def instrs_info(instrs, mark=None, range=None):
         if idx == mark:
             ret[-1] = "\033[31m" + ret[-1] + "\033[0m"
     return ret
+
+
+def calc_stack_effect(instr: Instruction, *, jump: bool | None = None) -> int:
+    """
+    Gets the stack effect of the given instruction.
+
+    Args:
+        instr: The instruction.
+
+    Returns:
+        The stack effect of the instruction.
+
+    """
+    if sys.version_info >= (3, 11):
```

Comment:
```suggestion
    if sys.version_info[:2] == (3, 11):
```

3.12 应该没问题了，docstring 里也说明下仅仅是为了修复 3.11 的问题吧，并加上 3.11 明显错误的固定 -1 的源码（因为没别的文档、讨论可以说明了，还是直接放源码吧）

https://github.com/python/cpython/blob/3.11/Python/compile.c#L1123-L1124